src/drivers/common/device.h

```diff
@@ -22,34 +21,28 @@
 
 #define STATIC_DEVMOD_ID -1
 
-struct device;
-struct dev_module;
-struct dev_operations;
 struct idesc;
-struct file;
 struct idesc_ops;
 
-struct device {
-	const struct dev_operations *dev_dops;
-	const struct idesc_ops *dev_iops;
-};
-
-struct dev_operations {
-	int  (*open)   (struct dev_module *mod, void *dev_priv);
-	int  (*probe)  (struct dev_module *mod, void *dev_priv);
-	void (*remove) (struct dev_module *mod);
-};
-
 struct dev_module {
-	struct file dev_file;
 	int    dev_id;
 	char   name[DEV_NAME_LEN];
-	struct device *device;
+
+	const struct idesc_ops *dev_iops;
+
+	struct idesc *(*open)  (struct dev_module *, void *);
+	int 		  (*close) (struct idesc *);
+
 	void  *dev_priv;
 };
 
-extern struct dev_module *dev_module_create(struct device *dev,
-	const char *name, void *privdata);
+extern struct dev_module *dev_module_create(
+	const char *name,
+	struct idesc * (*open)  (struct dev_module *, void *),
+	int 		   (*close) (struct idesc *),
+	const struct idesc_ops *dev_iops,
+	void *privdata
+);
```

Comment:
I think it looks better if `);` is left on the same line with the last argument.